src/components/EditPercentageForm/EditPercentageForm.tsx

```diff
@@ -0,0 +1,78 @@
+import { Button, Form } from '@taskany/bricks';
+import React, { ChangeEvent, FC, useCallback } from 'react';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+
+import { EditPercentageFormControl } from '../EditPercentageFormControl/EditPercentageFormControl';
+import { MembershipInfo } from '../../modules/userTypes';
+import { trpc } from '../../trpc/trpcClient';
+import { useUserMutations } from '../../modules/userHooks';
+import { UpdateMembershipPercentage, updateMembershipPercentageSchema } from '../../modules/userSchemas';
+
+import { tr } from './EditPercentageForm.i18n';
+
+interface EditPercentageFormProps {
+    membership: MembershipInfo;
+}
+
+export const EditPercentageForm: FC<EditPercentageFormProps> = ({ membership }) => {
+    const { updatePercentage } = useUserMutations();
+    const {
+        handleSubmit,
+        setValue,
+        setError,
+        clearErrors,
+        getValues,
+        formState: { errors },
+    } = useForm<UpdateMembershipPercentage>({
+        resolver: zodResolver(updateMembershipPercentageSchema),
+        defaultValues: {
+            membershipId: membership.id,
+            percentage: membership.percentage,
+        },
+    });
+
+    const availableMembershipQuery = trpc.user.getAvailableMembershipPercentage.useQuery(membership.userId, {
+        enabled: !!membership.userId,
+    });
+    const max = availableMembershipQuery.data ?? 100;
+
+    const onPercentageChange = useCallback(
+        (e: ChangeEvent<HTMLInputElement>) => {
+            const percentage = e.target.value === '' ? null : parseInt(e.target.value, 10);
+            setValue('percentage', percentage);
+            if (percentage && percentage > max && percentage > Number(membership.percentage)) {
+                setError('percentage', { message: tr('Maximum value is {max}', { max }) });
+            } else {
+                clearErrors('percentage');
+            }
+        },
+        [max, setValue, membership.percentage, setError, clearErrors],
+    );
+
+    const onSubmit = async (data: UpdateMembershipPercentage) => {
+        await updatePercentage(data);
+        setValue('percentage', membership.percentage);
```

Comment:
you have already set in at onPercentageChange..

do you think it can be changed?
also you can return response from updatePercentage and use it value - it will be clearly